protocol.md

```diff
@@ -162,6 +163,67 @@ Receive Firmware Name and Version (after query)
 N  END_SYSEX         (0xF7)
 ```
 
+Report supported Firmata features
+---
+
+Report which optional Firmata features are supported by the firmware and the version of each feature. 
+
+As described in the [Feature registry documentation](https://github.com/firmata/protocol/blob/master/feature-registry.md),
+Firmata features are split into 2 categories: *Core features* and *Optional features*. The feature
+query reports the optional features that are supported by the firmware as well as the version
+of each supported optional feature.
+
+The feature query reduces the need to define a new pin mode for every introduced feature. Going
+forward pin modes will only be defined and maintained for microcontroller specific features such as 
+digital and analog I/O, PWM, I2C, UART, SPI, DAC, etc. The use of pin modes for non microcontroller
```

Comment:
> A scenario I would like to enable, would be a teaching application that discovered a board connected to your computer and could interactively build out its capabilities and allow a person to tinker in a classroom setting with a point and click interface.

A few of these types of applications have been developed using Firmata over the years but have been limited in capability to Analog Input, PWM, digital I/O and Servos. They enabled a user to interact with a connected board using a GUI composed of buttons and sliders to control outputs as well as visual feedback for inputs connected to the board.

[NetLabToolKit](http://www.netlabtoolkit.org/) is one example of a Firmata-based application that has been used as a teaching tool at [Art Center](http://www.artcenter.edu/academics/undergraduate-degrees/interaction-design/overview.html) for several years.